libs/core/src/components/sage-tabs/sage-tabs.tsx

```diff
@@ -0,0 +1,161 @@
+    this.tabPanels = this.el.querySelectorAll('sage-tabpanel');
+  }
+
+  private passPropsToChildren = () => {
+    this.tabs.forEach(child => {
+      if (this.activeTab) { child['activeTab'] = this.activeTab.toString()};
+      if (this.componentId) {child['parentComponent'] = this.componentId.toString()};
+      if (this.variant) {child['variant'] = this.variant.toString()};
+    });
+    this.tabPanels.forEach(child => {
+      if (this.activeTab) { child['activeTab'] = this.activeTab.toString()};
+      if (this.componentId) {child['parentComponent'] = this.componentId.toString()};
+      if (this.variant) {child['variant'] = this.variant.toString()};
+    });
+  }
+
+  @Listen('keydown', {})
+  handleKeyDown(ev: KeyboardEvent){
+    const tabList = Array.from(this.el.querySelectorAll('[role="tab"]'));
+    const activeEl = this.getActiveElement();
+    const tabLocations = this.getTabLocations(tabList);
+    const firstTabNumber = 0;
+    const lastTabNumber = tabLocations.length - 1;
+
+    if (ev.key === 'ArrowLeft') {
+      if (activeEl.id === tabLocations[0].id) {
+        tabLocations[lastTabNumber].focus();
+        this.activeTab = tabLocations[lastTabNumber].id;
+      } else {
+        for (let i = 0; i < tabLocations.length; i += 1) {
+          if(tabLocations[i].id === activeEl.id ) {
+            tabLocations[i - 1].focus();
+            this.activeTab = tabLocations[i - 1].id;
+          }
+        }
+      }
+    }
+
+    if (ev.key === 'ArrowRight') {
+      if (activeEl.id === tabLocations[lastTabNumber].id) {
+        tabLocations[firstTabNumber].focus();
+        this.activeTab = tabLocations[firstTabNumber].id;
+      } else {
+        for (let i = 0; i < tabLocations.length; i += 1) {
+          if(tabLocations[i].id === activeEl.id ) {
+            tabLocations[i + 1].focus();
+            this.activeTab = tabLocations[i + 1].id;
+          }
+        }
+      }
+    }
+
+    if (ev.key === 'Home') {
+      if (activeEl.id != tabLocations[firstTabNumber].id) {
+        tabLocations[firstTabNumber].focus();
+        this.activeTab = tabLocations[firstTabNumber].id;
+      }
+    }
+
+    if (ev.key === 'End') {
+      if (activeEl.id != tabLocations[lastTabNumber].id) {
+        tabLocations[lastTabNumber].focus();
+        this.activeTab = tabLocations[lastTabNumber].id;
+      }
+    }
+  }
+
+  private classNames = () => {
+    let className = `sage-tabs`;
+    if (this.variant && this.variant != 'primary') {
+      const variantClassName = `sage-tabs--${this.variant}`;
+      className += ' ' + variantClassName;
+    }
+    return className;
+  };
+
+  private getTabLocations(tabList) {
+    const tabs = [];
+    for (let i = 0; i < tabList.length; i += 1) {
+      const tab = tabList[i];
+
+      tabs.push(tab);
+    }
+    return(tabs);
+  }
+
+  // Copied code - might be overkill or ideal to make global?
+  private getActiveElement(root: Document | ShadowRoot = document): Element | null {
```

Comment:
Why do you feel this might be overkill?

Reply:
Originally I thought this logic may be more than we needed as it checks for more than just shadowroot active elements but now that I've been using it I think its actually working well.